src/main/java/com/conveyal/datatools/manager/models/Project.java

```diff
@@ -105,6 +105,19 @@ public Project() {
                 .collect(Collectors.toList());
     }
 
+    /**
+     * Get all the labels for this project.
+     */
+    public Collection<Label> retrieveProjectLabels() {
+        return Persistence.labels.getAll().stream()
+                .filter(label -> this.id.equals(label.projectId))
+                .collect(Collectors.toList());
```

Comment:
Please make a query to the database that does the filtering within the database so that not all labels are fetched from the database.

Reply:
Good fix and is fewer lines as well! Have also made this change to filtering feed sources in 
85261da